woocommerce/woocommerce
# Email the site admin when a payment gateway is enabled

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





We now email the site admin when a payment gateway is enabled. 



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. checkout the branch locally and run `wp-env start` or spin up a JN site with this PR
2. configure the email system or install a plugin like "Mail Debug for WooCommerce"
3. activate and deactivate the existing payment gateways, ensure an email is sent to notify the admin (email from WordPress Settings / General) when a payment gateway is activated
4. install an additional payment gateway such as [WooPayments](https://wordpress.org/plugins/woocommerce-payments/) and ensure things work as expected for these, too
5. ensure you can add an email address to the notification email, e.g., like so:

```
function my_callback( $email_addresses ) {
	$email_addresses[] = 'me@example.com';
	return $email_addresses;	
}
add_action( 'wc_payment_gateway_enabled_notification_email_addresses', 'my_callback' );
```

6. ensure the `add_option_` event works as well as the `update_option_` event (e.g., via the `wp option delete woocommerce_cod_settings` you proposed earlier)
7. ensure a sent email has a corresponding log entry under WooCommerce / Status / Logs



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
